app/models/section.rb

```diff
@@ -159,14 +160,18 @@ def eql?(other)
     id.eql?(other.id)
   end
 
-protected
-
-  attr_reader :slug_generator, :edition_factory
-
   def never_published?
     !published?
   end
 
+  def change_note_required?
+    !(never_published? || minor_update?)
```

Comment:
Hmm. Change of plan. There's already a `Section#has_ever_been_published?` method which seems to be doing something different again. I think this is all a bit of a tangle and so I'm going to leave it for now.